### DISCUSSION TITLE
Why are the policy toggle parameters not booleans?

### DISCUSSION THREAD

**Discussion by JustinGrote** [Azure/Enterprise-Scale#670] [Q&A] [answered]
Before I create an issue, I was curious if there was a good reason why these are string with "Yes No" validation rather than just simply booleans.
https://github.com/Azure/Enterprise-Scale/blob/38f4f1ebe49a887dde67f589e1d2c5a184739819/docs/reference/contoso/armTemplates/auxiliary/identity.json#L12-L56

**Comment by jtracey93** [comment, marked as answer]
@krnese could you advise here? 👍 

I suspect its to do with the portal deployment experience.

**Reply by krnese** [reply]
Yes, this is for the curated portal experience. What's the issue?

**Reply by JustinGrote** [reply]
Just seems like this would be better suited with a boolean and a checkbox, but I suppose it comes down to a stylistic choice without being a technical constrant.

https://docs.microsoft.com/en-us/azure/azure-resource-manager/managed-applications/microsoft-common-checkbox?WT.mc_id=Portal-Microsoft_Azure_CreateUIDef

**Reply by krnese** [reply]
With checkboxes, we would not be able to default to what we would recommend, and require user to explicitly opt-in for the various options we provide. 

Further, we would lose some flexibility in terms of how we handle certain conditions if we used bool as type.
